src/routes/maven.rs

```diff
@@ -0,0 +1,262 @@
+
+                    let mod_exists = sqlx::query!(
+                        "SELECT EXISTS(SELECT 1 FROM team_members WHERE team_id = $1 AND user_id = $2)",
+                        data.inner.team_id as database::models::ids::TeamId,
+                        user_id as database::models::ids::UserId,
+                    )
+                    .fetch_one(&**pool)
+                    .await
+                    .map_err(|e| ApiError::DatabaseError(e.into()))?
+                    .exists;
+
+                    authorized = mod_exists.unwrap_or(false);
+                }
+            }
+        }
+
+        if authorized {
+            let version_names = sqlx::query!(
+                "
+                SELECT version_number, release_channels.channel channel
+                FROM versions
+                LEFT JOIN release_channels ON release_channels.id = versions.release_channel
+                WHERE mod_id = $1
+                ",
+                data.inner.id as database::models::ids::ModId
+            )
+            .fetch_all(&**pool)
+            .await
+            .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+            let respdata = Metadata {
+                group_id: "maven.modrinth".to_string(),
+                artifact_id: string,
+                versioning: Versioning {
+                    latest: version_names
+                        .last()
+                        .map_or("release", |x| &x.version_number)
+                        .to_string(),
+                    release: version_names
+                        .iter()
+                        .rfind(|x| x.channel == "release")
+                        .map_or("", |x| &x.version_number)
+                        .to_string(),
+                    versions: Versions {
+                        versions: version_names
+                            .iter()
+                            .map(|x| x.version_number.clone())
+                            .collect::<Vec<_>>(),
+                    },
+                    last_updated: data.inner.updated.format("%Y%m%d%H%M%S").to_string(),
+                },
+            };
+
+            // return Ok(HttpResponse::Ok().content_type("text/xml").body(response));
+            Ok(HttpResponse::Ok()
+                .content_type("text/xml")
+                .body(yaserde::ser::to_string(&respdata).map_err(|e| ApiError::XmlError(e))?))
+        } else {
+            Ok(HttpResponse::NotFound().body(""))
+        }
+    } else {
+        Ok(HttpResponse::NotFound().body(""))
+    }
+}
+
+#[get("maven/modrinth/{id}/{versionnum}/{file}")]
+pub async fn version_file(
+    req: HttpRequest,
+    web::Path((string, vnum, file)): web::Path<(String, String, String)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let id_option: Option<ModId> = serde_json::from_str(&*format!("\"{}\"", string)).ok();
+
+    let mut mod_data;
+
+    if let Some(id) = id_option {
```

Comment:
You do not even need to have a mutable variable here, rust allows this kind of code to run:
```rust
let data = if request.is_ok() {
  request.data
} else {
  Data::new()
}
```